inc/namespace.php

```diff
@@ -219,3 +253,80 @@ function ajax_query_attachments() : void {
 
 	wp_send_json_success( $items->toArray() );
 }
+
+function match_attachments_for_js( array $response, WP_Post $attachment ) : array {
+	global $amf_inserted;
+
+	if ( empty( $amf_inserted[ $attachment->guid ] ) ) {
+		return new WP_Error( 'amf_error', __( 'Unable to prepare attachment response.', 'asset-manager-framework' ) );
+	}
+
+	if ( strpos( $attachment->post_name, 'amf-' ) === 0 ) {
+		// Correct the image sizes array to remove the duplicated base URL when a
+		// full URL is provided as the size file name.
+		if ( ! empty( $response['sizes'] ) ) {
+			$base_url = str_replace( wp_basename( $attachment->guid ), '', $attachment->guid );
+			foreach ( $response['sizes'] as $name => $size ) {
+				if ( mb_substr_count( $size['url'], $base_url ) < 2 ) {
+					continue;
+				}
+				$response['sizes'][ $name ]['url'] = $base_url . str_replace( $base_url, '', $size['url'] );
+			}
+		}
+
+		return $response;
+	}
+
+	return wp_prepare_attachment_for_js( $amf_inserted[ $attachment->guid ] );
+}
+
+function handle_upload( $_, $file ) {
+	global $amf_upload;
+
+	$provider = get_provider();
+
+	// Reset result of upload.
+	$amf_upload = null;
+
+	try {
+		$upload = new FileUpload( $file );
+		$media = $provider->handle_upload( $upload );
+
+		$amf_upload = [
+			'file' => $upload->tmp_name,
+			'url' => $media->url,
+			'type' => $media->mime,
+		];
+
+		// Hook into the subsequent wp_insert_attachment() call to remove the
+		// new unwanted local post on shutdown.
+		add_action( 'add_attachment', __NAMESPACE__ . '\\remove_local_attachment' );
```

Comment:
I assume there is no way to actually prevent the thing we eventually undo from happening in the first place...?

Reply:
Nope, pretty much every core function will run `wp_insert_attachment` after any upload with no way to skip it. At least that I could see but a fresh pair of eyes couldn't hurt.